NEW get api jwt token

frederic34 opened the pull request:
php-jwt version is 6.10.0 because it's the last version that support php 7.4
- [x] generate token
- [x] generate refresh token
- [x] make token usable by api
- [x] can refresh token using refresh token
- [x] waiting for comments

frederic34 committed:
NEW get jwt token for api use

frederic34 committed:
NEW get jwt token for api use

frederic34 committed:
NEW get jwt token for api use

frederic34 committed:
NEW get jwt token for api use

frederic34 committed:
NEW get jwt token for api use

mdeweerd commented on .gitleaks.toml:
```diff
@@ -0,0 +1,5 @@
+[allowlist]
+    # Exclure un fichier spécifique par son chemin
+    paths = [
+        '''^htdocs/includes/php-jwt/README.md$''',
```
In short:
This looks ok but the following is somewhat more generic
'''(?:^|/)htdocs/includes/php-jwt/README\ .md$'''`,` 

And `[[allowlists]]` would be the appropriate section now.
allowlist(s) do seem to be cumulative (they add to the existing list),

I would do a small local test on a branch with a secret in a file that should still be checked.

## Discussion:

The analysis of https://github.com/gitleaks/gitleaks/blob/master/config/gitleaks.toml shows these are indeed regexes.

The main question is : de .gitleaks.toml extend this list or replace it.
Possibly a '.gitleaksignore` file is more appropriate.

The readme of gitleaks indicates:
`# ⚠️ In v8.25.0 `[allowlist]` was replaced with '[[allowlists]]'.`


My guess is that `[[allowlists]]` can be specified multiple times and `[allowlist]` only once.  Therefore the latter might replace the default list, while the former probably adds to the list.   But the gitleaks readme indicates `Allowlist arrays are appended and can contain duplicates.`




Other observations:
So dots must be escaped.

At the start of the path they use `(?:^|/)` - I do not think this is needed here (the example is a generic configuration).

frederic34 committed:
NEW get jwt token for api use

frederic34 committed:
Merge remote-tracking branch 'upstream/develop' into api_jwt_token

frederic34 committed:
wip

mdeweerd commented on .gitleaks.toml:
```diff
@@ -0,0 +1,5 @@
+[[allowlists]]
+    # Exclude a specific file
+    paths = [
+        '''(?:^|/)htdocs/includes/php-jwt/README\ .md$''',
```
The `\` is for the dot, the space should be removed.

frederic34 replied:
aïe

frederic34 committed:
wip

frederic34 committed:
wip

frederic34 committed:
wip

frederic34 committed:
wip

frederic34 committed:
Merge remote-tracking branch 'upstream/develop' into api_jwt_token

frederic34 committed:
wip

frederic34 committed:
wip

frederic34 committed:
wip

frederic34 committed:
wip

frederic34 committed:
wip

frederic34 committed:
wip

wip

wip

eldy committed:
Trans

W1W1-M committed:
Fixed return object for expense report API (#35813)

github-actions[bot] committed:
PHPStan > Update baseline (#35810)

Co-authored-by: Dolibot <dolibarr-bot@users.noreply.github.com>

joseplluis committed:
NEW Add configuration for default timesheet menu (#35805)

* Add configuration for default timesheet menu

Using PROJECT_OPEN_ALWAYS_ON_TIMESHEET config, you can use variables "perweek" (default), "perday" or "permonth"

* Update eldy.lib.php

* Fix default value for optionTimesheet variable

---------

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

W1W1-M committed:
NEW: Option to clone parent categories on variant creation (#35806)

* Added option to clone parent categories on product variant creation

* Updated PHPDoc

vmaury committed:
Display error message when impossible to create user from contact (#35807)

Co-authored-by: vmaury <vmaury@vmaury-Lafite-Pro-16-AMD>

wip

frederic34 committed:
fix

frederic34 committed:
Merge remote-tracking branch 'upstream/develop' into api_jwt_token

frederic34 committed:
fix

frederic34 committed:
Merge remote-tracking branch 'upstream/develop' into api_jwt_token

frederic34 committed:
Merge remote-tracking branch 'upstream/develop' into api_jwt_token

frederic34 committed:
fix

frederic34 committed:
fix

frederic34 committed:
fix

frederic34 committed:
clean baseline

frederic34 committed:
get a new token with refresh token

frederic34 committed:
get a new token with refresh token

frederic34 committed:
get a new token with refresh token

frederic34 committed:
Merge remote-tracking branch 'upstream/develop' into api_jwt_token

frederic34 committed:
get a new token with refresh token

frederic34 committed:
get a new token with refresh token

frederic34 committed:
get a new token with refresh token

frederic34 committed:
Merge remote-tracking branch 'upstream/develop' into api_jwt_token

frederic34 committed:
get a new token with refresh token

frederic34 committed:
get a new token with refresh token

frederic34 committed:
get a new token with refresh token

frederic34 committed:
get a new token with refresh token

frederic34 committed:
Merge remote-tracking branch 'upstream/develop' into api_jwt_token

frederic34 committed:
get a new token with refresh token

frederic34 committed:
check permission

frederic34 committed:
Merge remote-tracking branch 'upstream/develop' into api_jwt_token

frederic34 committed:
check permission

frederic34 committed:
Merge branch 'develop' into api_jwt_token

frederic34 committed:
Merge branch 'develop' into api_jwt_token

frederic34 committed:
Merge branch 'develop' into api_jwt_token

frederic34 committed:
Merge remote-tracking branch 'upstream/develop' into api_jwt_token

frederic34 committed:
Merge remote-tracking branch 'origin/api_jwt_token' into api_jwt_token

frederic34 committed:
Merge branch 'develop' into api_jwt_token

frederic34 committed:
Merge branch 'develop' into api_jwt_token

frederic34 committed:
Merge branch 'develop' into api_jwt_token
